test/gtest/uct/test_md.cc

```diff
@@ -82,12 +82,29 @@ test_md::test_md()
 
 void test_md::init()
 {
+    const std::vector<ucs_memory_type_t>
+        supported_mem_types = mem_buffer::supported_mem_types();
     ucs::test_base::init();
     UCS_TEST_CREATE_HANDLE(uct_md_h, m_md, uct_md_close, uct_md_open,
                            GetParam().component, GetParam().md_name.c_str(),
                            m_md_config);
 
     ucs_status_t status = uct_md_query(m_md, &m_md_attr);
+
+    /* Set context to allow device allocations through uct_mem_alloc */
+
+
+    for (std::vector<ucs_memory_type_t>::const_iterator
+         iter = supported_mem_types.begin();
+         iter != supported_mem_types.end(); ++iter) {
+         if (m_md_attr.cap.alloc_mem_types & UCS_BIT(*iter)) {
+            if ((*iter == UCS_MEMORY_TYPE_CUDA) ||
+                (*iter == UCS_MEMORY_TYPE_CUDA_MANAGED)) {
+                void *address = mem_buffer::allocate(1, (ucs_memory_type_t)*iter);
+                mem_buffer::release(address, (ucs_memory_type_t)*iter);
+            }
+        }
+    }
```

Comment:
can we simplify with  mem_buffer::set_context() and move this code(or replace with cudaSetDevice())  to mem_buffer class? and re-use this in test_mem.cc also

Reply:
Can try this now.